src/app/util/attribute-storage.cpp

```diff
@@ -638,10 +638,29 @@ EmberAfStatus emAfReadOrWriteAttribute(EmberAfAttributeSearchRecord * attRecord,
                                 // Is the attribute externally stored?
                                 if (am->mask & ATTRIBUTE_MASK_EXTERNAL_STORAGE)
                                 {
-                                    return (write ? emberAfExternalAttributeWriteCallback(attRecord->endpoint, attRecord->clusterId,
-                                                                                          am, buffer)
-                                                  : emberAfExternalAttributeReadCallback(attRecord->endpoint, attRecord->clusterId,
-                                                                                         am, buffer, emberAfAttributeSize(am)));
+                                    if (write) 
+                                    {
+                                        return emberAfExternalAttributeWriteCallback(attRecord->endpoint, attRecord->clusterId, am, buffer);
+                                    } 
+                                    else
+                                    {
+                                        if ((readLength < emberAfAttributeSize(am)) && (readLength != 0))
+                                        {
+                                            if (emberAfIsStringAttributeType(am->attributeType) || emberAfIsLongStringAttributeType(am->attributeType))
+                                            {
+                                                ChipLogError(DataManagement, "Attribute string to large for buffer, the string will be truncated. ReadLength: %u, < AttributeSize: %u", readLength, emberAfAttributeSize(am));
```

Comment:
This log message is not necessarily true. If the target only wants to put in a small string, it is not truncated. 
The question is: how should we reword:

Something like: "attribute size exceeds maximum read size"?